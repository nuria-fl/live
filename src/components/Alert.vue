<template>
  <modal
    :is-closeable="true"
    :visible.sync="visible"
    @update:visible="closeModal"
  >
    <p slot="body">
      {{ text }}
    </p>
  </modal>
</template>

<script>
import Modal from '@/components/Modal'
import eventBus from '@/utils/eventBus'

export default {
  components: {
    Modal
  },
  data() {
    return {
      visible: false,
      text: ''
    }
  },
  mounted() {
    eventBus.$on('showModal', this.showModal)
  },
  methods: {
    showModal(data) {
      this.visible = true
      this.text = data.body
    },
    closeModal() {
      this.text = ''
    }
  }
}
</script>

<style lang="css"></style>
