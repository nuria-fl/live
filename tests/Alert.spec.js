import { mount } from 'vue-test-utils'

import Alert from '../src/components/Alert'

describe('Alert snapshot', () => {
  let wrapper

  beforeEach(() => {
    wrapper = mount(Alert)
  })

  test('should match snapshot', () => {
    wrapper.setData({visible: true, text: 'hi'})

    expect(wrapper.vm.$el).toMatchSnapshot()
  })
})
