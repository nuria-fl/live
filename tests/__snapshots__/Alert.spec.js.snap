// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert snapshot should match snapshot 1`] = `
<div
  class="Modal"
>
  <div
    class="Modal__content"
  >
    <header />
     
    <section>
      <p>
        hi
      </p>
    </section>
     
    <footer>
      <button
        class="Btn"
      >
        
        Ok
      
      </button>
       
    </footer>
  </div>
</div>
`;
